### PR TITLE
[SW, C] Clean up

### PR DESCRIPTION
**Pull request by engdoreis**
This PR:
- Clean some remaining cmake files.
- Add a clang-format rule.
- Format the existing code base.
- Add a linting step to the github actions.

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -31,6 +31,8 @@ jobs:
         wget https://github.com/lowRISC/lowrisc-toolchains/releases/download/20230427-1/lowrisc-toolchain-gcc-rv32imcb-20230427-1.tar.xz
         tar -xvf lowrisc-toolchain-gcc-rv32imcb-20230427-1.tar.xz
         echo "/tmp/lowrisc-toolchain-gcc-rv32imcb-20230427-1/bin" >> $GITHUB_PATH
+    - name: Linting
+      run: find -name "*.[c|h]" -exec clang-format -n {} \;
```
Comment (engdoreis) [inline .github/workflows/cmake.yml:35]: Is there a batter way?

**Review by marnovandermaas — APPROVED** [review]
Looks good to me. Thanks for adding this!

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (jwnrt) [reply .github/workflows/cmake.yml:35]: This seems okay to me

**Review by jwnrt — APPROVED** [review]

File: sw/c/.clang-format
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/c/.clang-format:1]: VSCode is able to pick it up, so I'll keep it in the C dir.

**Pull request merged**